tests/test_config_flow.py

```diff
@@ -0,0 +1,163 @@
+"""Test the Somfy TaHoma config flow."""
+from asynctest import patch
+from custom_components.tahoma import config_flow
+from pyhoma.exceptions import BadCredentialsException, TooManyRequestsException
+
+from homeassistant import config_entries, setup
+
+from .common import MockConfigEntry
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch("pyhoma.client.TahomaClient.login", return_value=True), patch(
+        "custom_components.tahoma.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "custom_components.tahoma.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"username": "test@testdomain.com", "password": "test-password"},
+        )
+
+    assert result2["type"] == "create_entry"
```

Comment:
You can do one assert:
```Python
assert result2 == {"type": "create_entry", "title": "test@testdomain.com", ...}
```
except if their is some random value in result2.

Reply:
I prefer to match them on a per key base to easily understand which part is failing and I also believe that we can not change the order in your example, without breaking the test?

Does this make sense?

Reply:
Pytest have a nice output with the json assertion. It does not care about the order (except if an array). Have a try ;)

Reply:
```
E       AssertionError: assert {'data': {'pa...8c074b1', ...} == {'data': {'pa...create_entry'}
E         Omitting 2 identical items, use -vv to show
E         Differing items:
E         {'data': {'password': 'test-password', 'username': 'test@testdomain.com'}} != {'data': {'password': '2test-password', 'username': 'test@testdomain.com'}}
E         Left contains 6 more items:
E         {'description': None,
E          'description_placeholders': None,
E          'flow_id': 'cb53d0783e9b40768027187808c074b1',...
E         
```

There are more items in the list apparently, which is why the assert doesn't work otherwise. This is also how they do it in the other components.